services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/helpers/FlowHelper.groovy

```diff
@@ -108,7 +109,39 @@ class FlowHelper {
         log.debug("Deleting flow '$flowId'")
         def response = northbound.deleteFlow(flowId)
 
-        checkRulesOnSwitches(flowEntry, RULES_DELETION_TIME, false)
+        def switches = [flowEntry.left.sourceSwitch, flowEntry.left.destinationSwitch].toSet()
+        def cookies = [flowEntry.left.cookie, flowEntry.right.cookie]
+        checkRulesOnSwitches(switches, cookies, RULES_DELETION_TIME, false)
+
+        return response
+    }
+
+    /**
+     * Updates flow with checking flow status and rules on source and destination switches.
+     * It is supposed if rules are installed on source and destination switches, the flow is completely updated.
+     */
+    FlowPayload updateFlow(String flowId, FlowPayload flow) {
+        def flowEntryBeforeUpdate = db.getFlow(flowId)
+
+        log.debug("Updating flow '${flow.id}'")
+        def response = northbound.updateFlow(flowId, flow)
+        Wrappers.wait(WAIT_OFFSET) { assert northbound.getFlowStatus(flow.id).status == FlowState.UP }
+
+        def flowEntryAfterUpdate = db.getFlow(flowId)
+
+        if (flowEntryBeforeUpdate.left.sourceSwitch != flowEntryAfterUpdate.left.sourceSwitch) {
```

Comment:
After update the flow will get new cookies regardless of whether the switch was changed, so you can remove `if` condition. Another thing is that today we released the multi-stage rules installation process, so I guess Kilda should take care of installing new rules first before removing the old ones (you can assume this in your code and check only the latest operation)

Reply:
Let's keep all checks until we add a new test to verify this functionality.